src/PeerDigest.h

```diff
@@ -49,7 +49,7 @@ class DigestFetchState
     DigestFetchState(PeerDigest *,HttpRequest *);
     ~DigestFetchState();
 
-    PeerDigest *pd;
+    CbcPointer<PeerDigest> pd;
```

Comment:
If possible, please make this a `RefCount` instead of `CbcPointer`. Then you can replace all `delete fetch` with `fetch = nullptr; // refcounted` and not worry about the memory management at all.

Reply:
@eduard, I am aware. The issues is that a number of old pointers used CBDATA for pseudo-reference counting and memory management - when they should be using the simpler RefCount and MEMPROXY. I was suspicious of this being another one of those cases - however Alex has pointed out that CBDATA behaviour is still needed.

@rousskov. Thank you for highlighting that.